partition/bootloader_test.go

```diff
@@ -157,3 +161,23 @@ func (s *PartitionTestSuite) TestInstallBootloaderConfig(c *C) {
 		c.Assert(osutil.FileExists(fn), Equals, true)
 	}
 }
+
+func (s *PartitionTestSuite) TestRebootNoError(c *C) {
+	systemdCmd := testutil.MockCommand(c, "shutdown", `
+exit 0
+	`)
+	defer systemdCmd.Restore()
+
+	err := reboot(10)
+	c.Assert(err, IsNil)
+}
+
+func (s *PartitionTestSuite) TestRebootError(c *C) {
+	systemdCmd := testutil.MockCommand(c, "shutdown", `
+exit 1
+	`)
+	defer systemdCmd.Restore()
+
+	err := reboot(10)
+	c.Assert(err, NotNil)
```

Comment:
If we could use `c.Assert(err, ErrorMatches, "the actual error")` that would be great.